platform/openide.util.ui/src/org/openide/util/Utilities.java

```diff
@@ -1357,17 +1357,50 @@ private static Frame findMainWindow()
      */
     // PR4739
     public static Component findDialogParent() {
-        Component parent = KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusOwner();
+        return findDialogParent(null);
+    }
+
+    /**
+     * Finds an appropriate component to use for a dialog's parent. Similar to {@link #findDialogParent()}
+     * with the ability to specify a suggested parent component.
+     *
+     * @param suggestedParent the component to return if suitable
+     * @return A suitable parent component for swing dialog displayers.
```

Comment:
Seems like this is for any dialog, not "dialog displayer". Maybe I'm just confusing myself.

Is it useful to mention that a suggestedParent is unsuitable if there's a modal dialog active and the suggestParent is not it.

Reply:
Sure, I was just using the same documentation in the existing `findDialogParent()` method. How about this (based on the added fallback supplier I [suggested earlier](https://github.com/apache/netbeans/pull/5989#discussion_r1206909445))?

```java
/**
 * Finds an appropriate component to use for a modal dialog's parent. Similar to {@link #findDialogParent()}
 * with the ability to specify a suggested parent component and fallback.
 *
 * @param suggestedParent
 *            the component to return if non-null and valid
 * @param fallback
 *            a supplier for the parent if no other suitable candidate exists
 * @return the suggested parent if there is either no active modal dialog or it is contained
 *         within that dialog, otherwise the active modal dialog or the fallback.
 */
```